crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -4516,7 +4551,7 @@ mod tests {
         )
         .unwrap();
 
-        assert_public_ty(&db, "src/a.py", "y", "Unbound | Literal[2, 3]");
+        assert_public_ty(&db, "src/a.py", "y", "Literal[2, 3]");
```

Comment:
same as above, should have an assertion in some form that `y` can be unbound here